File: src/atcoder/abc317/swift/abc317_a.swift
func getv() -> [Int] { readLine()!.split(separator: " ").map { Int($0)! } }
func get3() -> (Int, Int, Int) {
  let vs = getv()
  return (vs[0], vs[1], vs[2])
}

struct Potion {
  let id: Int
  let hp: Int
}

let (n, h, x) = get3()
let potions = zip(0..<n, getv()).map { (i, v) in Potion(id: i + 1, hp: v) }
let ans = potions.filter { x <= h + $0.hp }.first!

print(ans.id)

File: src/atcoder/abc317/scala/abc317_a.scala
def getv(): Array[Int] = io.StdIn.readLine().split(" ").map(_.toInt)
def get3(): (Int, Int, Int) =
  val v = getv()
  return (v(0), v(1), v(2))

case class Potion(id: Int, hp: Int)

object Main extends App:
  val (n, h, x) = get3()
  val potions = getv().zipWithIndex.map((hp, i) => Potion(i+1, hp))
  val ans = potions.filter(x <= h + _.hp).head
  println(ans.id)
